sdk/core/azure-core/test/ut/authorization_challenge_parser_test.cpp

```diff
@@ -205,3 +207,57 @@ TEST(AuthorizationChallengeParser, StorageChallenge)
+{
+  EXPECT_FALSE((ClassTraits<AuthorizationChallengeParser>::is_constructible()));
+  EXPECT_FALSE((ClassTraits<AuthorizationChallengeParser>::is_trivially_constructible()));
+  EXPECT_FALSE((ClassTraits<AuthorizationChallengeParser>::is_nothrow_constructible()));
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_default_constructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_trivially_default_constructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_nothrow_default_constructible());
+}
+
+TEST(AuthorizationChallengeParser, Destructible)
+{
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_destructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_trivially_destructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_nothrow_destructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::has_virtual_destructor());
+}
+
+TEST(AuthorizationChallengeParser, CopyAndMoveConstructible)
+{
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_copy_constructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_trivially_copy_constructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_nothrow_copy_constructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_move_constructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_trivially_move_constructible());
+  EXPECT_FALSE(ClassTraits<AuthorizationChallengeParser>::is_nothrow_move_constructible());
+}
+
+TEST(AuthorizationChallengeParser, Assignable)
+{
+  EXPECT_TRUE(
+      ClassTraits<AuthorizationChallengeParser>::is_assignable<AuthorizationChallengeParser>());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_assignable<
+              const AuthorizationChallengeParser>());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_trivially_assignable<
+              AuthorizationChallengeParser>());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_trivially_assignable<
+              const AuthorizationChallengeParser>());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_nothrow_assignable<
+              AuthorizationChallengeParser>());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_nothrow_assignable<
+              const AuthorizationChallengeParser>());
+}
+
+TEST(AuthorizationChallengeParser, CopyAndMoveAssignable)
+{
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_copy_assignable());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_trivially_copy_assignable());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_nothrow_copy_assignable());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_move_assignable());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_trivially_move_assignable());
+  EXPECT_TRUE(ClassTraits<AuthorizationChallengeParser>::is_nothrow_move_assignable());
+}
```

Comment:
Trailing crlf (several of them in this code review)

Reply:
Otherwise this looks fine.

Reply:
Thanks! and oh note that I have made the decision about which constructor contracts to test for based on an examination of the usage patterns of the classes. Please feel free to point out any instance where I might have mis-judged the intent.